pupil-spa/src/app/check/check.component.ts

```diff
@@ -80,10 +84,29 @@ export class CheckComponent implements OnInit {
       history.go(1);
     };
 
-    this.state = 0;
-    this.isWarmUp = true;
-    this.viewState = 'warmup-intro';
-    this.totalNumberOfQuestions = this.warmupQuestionService.getNumberOfQuestions();
+    // set up the state
+    if (this.storageService.getItem(CheckComponent.checkStateKey)) {
+      // existing check state detected
+      // assume we are reloading during a check
+      const existingState = this.storageService.getItem(CheckComponent.checkStateKey);
+      if (!this.isValidState(existingState)) {
```

Comment:
@jon-shipley not a big fan of nested if as they oppose the principles of clean coding.
How about you approach it with a gateway method?
https://wpshout.com/unconditionally-refactoring-nested-statements-cleaner-code/

Reply:
if (!this.storageService.getItem(CheckComponent.checkStateKey)) {
..code 
return
}

Reply:
@GeorgeChatzigiannis  I have refactored the code block to make it clearer.